app/components/dex-create.jsx

```diff
@@ -66,7 +81,7 @@ export class DexCreate extends Component {
 
   render () {
     const { isOpen, session } = this.props;
-    const { error, game, regional, url } = this.state;
+    const { error, game, games, regional, url } = this.state;
 
```

Comment:
the fact that the whole component is being rendered when now, there are parts that need to be loaded async (i.e. the games), is proving to be problematic. cause when this `render` function is first called, `games` is `undefined` because the api call hasnt finished yet. so youll get the error `Cannot read property 'map' of undefined`. so we need to prevent the rest of the component from rendering unnecessarily. i think we can just accomplish this with

```js
if (!isOpen || !games) {
  return null;
}
```

this should be fine for us cause its just a precaution since were loading games from the profile and itll take some time before someone clicks the edit of create button. the alternative is to have a `loading` variable and a loading state and stuff. kinda think its overkill here but ill leave it to your discretion. i think we might need it for the register tho since the results of the fact that the games arent there will be apparent, again, ill leave it to you

Reply:
makes sense - i think it's fine here but we should add one to register.

Reply:
ok nvm it's loading very quickly, so not gonna add a loading state to register either 😂